src/VisualStudio/Core/Def/Implementation/TableDataSource/AbstractRoslynTableDataSource.cs

```diff
@@ -14,8 +14,12 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation.TableDataSource
     /// </summary>
     internal abstract class AbstractRoslynTableDataSource<TData> : AbstractTableDataSource<TData>
     {
-        public AbstractRoslynTableDataSource(Workspace workspace) : base(workspace)
+        private readonly ProgressReporter _reporterOpt;
+
+        public AbstractRoslynTableDataSource(Workspace workspace, ProgressReporter reporterOpt = null) : base(workspace)
```

Comment:
It seems strange that this is optional.  I would want subclasses to think if they should provide this or not explicitly.